harness/determined/common/experimental/trial.py

```diff
@@ -233,10 +233,13 @@ def get_trial_checkpoints(offset: int) -> bindings.v1GetTrialCheckpointsResponse
                 else None,
                 sortByMetric=sort_by if isinstance(sort_by, str) else None,
                 offset=offset,
-                limit=limit,
+                limit=max_results,
             )
 
-        resps = api.read_paginated(get_trial_checkpoints)
+        resps = api.read_paginated(
```

Comment:
This sets the ceiling, but doesn't guarantee enough results are returned. If `max_results` aren't returned with the REST API's first page, we won't end up with enough returned checkpoints.

I think maybe `api.read_paginated` is the right place to put this functionality.

Reply:
Well, at least in the code, `limit` is [respected when passed](https://github.com/determined-ai/determined/blob/main/master/internal/api/params.go#L60). The REST API will never return less than `limit` if it has enough items.

Reply:
Ok. This looks good. Still slight preference that `max_results` is implemented in `read_paginated`, but this works and was easy. Whatever you think is best.